src/sql/workbench/contrib/notebook/browser/cellViews/code.component.ts

```diff
@@ -446,4 +458,75 @@ export class CodeComponent extends CellView implements OnInit, OnChanges {
 			}
 		}
 	}
+
+	public onCellLanguageClick(event: any): void {
+		this._notebookService.getSupportedLanguagesForProvider(this._model.providerId, this._model.selectedKernelDisplayName)
+			.then(languages => this.pickCellLanguage(languages))
+			.then(selection => {
+				if (selection?.languageId) {
+					this._cellModel.setOverrideLanguage(selection.languageId);
+				}
+			})
+			.catch(err => onUnexpectedError(err));
+	}
+
+	private pickCellLanguage(languages: string[]): Promise<ILanguagePickInput | undefined> {
```

Comment:
Most of the logic in this method (as well as the getFakeResource method) was copied from VS Code's existing cell language implementation in coreActions.ts: https://github.com/microsoft/azuredatastudio/blob/main/src/vs/workbench/contrib/notebook/browser/contrib/coreActions.ts#L1727